diff --git a/hterm/js/hterm_scrollport.js b/hterm/js/hterm_scrollport.js
index 8ca07d04..b83b95ca 100644
--- a/hterm/js/hterm_scrollport.js
+++ b/hterm/js/hterm_scrollport.js
@@ -545,13 +545,7 @@ hterm.ScrollPort.prototype.decorate = function(div, callback) {
     }
   };
 
-  // Insert Iframe content asynchronously in FF.  Otherwise when the frame's
-  // load event fires in FF it clears out the content of the iframe.
-  if ('mozInnerScreenX' in globalThis) { // detect a FF only property
-    this.iframe_.addEventListener('load', () => onLoad());
-  } else {
-    onLoad();
-  }
+  onLoad();
 };
 
 
